tests/integration/network/anvil/test_network_env.py

```diff
@@ -73,9 +73,38 @@ def test_failed_transaction():
 # XXX: probably want to test deployment revert behavior
 
 
-def test_deployment_db():
+def test_deployment_db_overriden_contract_name():
     with set_deployments_db(DeploymentsDB(":memory:")) as db:
         arg = 5
+        contract_name = "test_deployment"
+
+        # contract is written to deployments db
+        contract = boa.loads(code, arg, contract_name=contract_name)
+
+        # test get_deployments()
+        deployment = next(db.get_deployments())
+
+        initcode = contract.compiler_data.bytecode + arg.to_bytes(32, "big")
+
+        # sanity check all the fields
+        assert deployment.contract_address == contract.address
```

Comment:
no need to change now, however I usually prefer comparing the whole thing at once, e.g. with `asdict`. It leads to much better to understand errors imo